torch/csrc/jit/script/compiler.cpp

```diff
@@ -118,6 +118,52 @@ struct CastValue : public SugaredValue {
   TypePtr type;
 };
 
+// expressions like x.int()
+struct ScalarTypeCastValue : public SugaredValue {
+  ScalarTypeCastValue(std::string type_str, NamedValue self)
+  : type_str(type_str), self(self) {}
+  std::string kind() const override {
+    std::stringstream ss;
+    ss << "<" << type_str << " scalar cast primitive>";
+    return ss.str();
+  }
+  static const std::unordered_map<std::string, std::string> &method_to_type_map() {
+    static std::unordered_map<std::string, std::string> method_to_type = {
+      {"byte", "_cast_Byte"},
+      {"char", "_cast_Char"},
+      {"double", "_cast_Double"},
+      {"float", "_cast_Float"},
+      {"int", "_cast_Int"},
+      {"long", "_cast_Long"},
+      {"short", "_cast_Short"},
+      {"half", "_cast_Half"}
+    };
+    return method_to_type;
+  }
+  std::shared_ptr<SugaredValue> call(
+      SourceRange loc,
+      Method & m,
+      at::ArrayRef<NamedValue> inputs,
+      at::ArrayRef<NamedValue> attributes,
+      size_t n_binders) override {
+    if (!attributes.empty())
```

Comment:
This error checking code is redundant with the error checking code in emitBuiltinCall.